src/components/Menu/Menu/Menu.tsx

```diff
@@ -104,6 +90,45 @@ const Menu: VibeComponent<MenuProps> & {
       });
     }, [originalChildren]);
 
+    useEffect(() => {
+      if (!id) {
+        console.warn("Menu should have a valid id prop");
+      }
+    }, [id]);
+
+    const updateActiveItemIndex = useCallback(
+      (index: number) => {
+        setActiveItemIndex(index);
+
+        const activeChild = children[index];
+        const ariaActiveDescendant = React.isValidElement(activeChild)
+          ? activeChild?.props?.id || `${id}-${index}`
```

Comment:
The only problematic thing here is that we define the default id of menu item twice: here and in the menu item component (`${id}-${index}`), so I think it will be the best to set the menu item default id as the default value (and then we will be better to remove `${id}-${index}` from here).

Because this is the only comment I have, I am approving the pr for not be a blocker.